kore/src/Kore/Step/Strategy.hs

```diff
@@ -291,14 +296,17 @@ insEdge (fromNode, toNode) exe@ExecutionGraph { graph } =
     exe { graph = Graph.insEdge (fromNode, toNode, mempty) graph }
 
 insChildNode
-    :: MonadState (Gr config (Seq rule)) m
+    :: (NFData config, NFData rule)
+    => MonadState (Gr config (Seq rule)) m
     => ChildNode config rule
     -> m Graph.Node
-insChildNode configNode =
-    State.state insChildNodeWorker
+insChildNode childNode = do
+    graph <- deepseq childNode State.get
+    let (node', graph') = insChildNodeWorker graph
+    deepseq graph' (State.put graph')
```

Comment:
Just to be sure I understand, this is what you were saying in the meeting about fully evaluating the graph when adding a new node, similar to calling `graph` after each step?